src/index.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * Type for (x, y) coordinate pair.
+ */
+ export type Point = { x: number, y: number };
+
+ /**
+  * Method signature for distance calculation.
+  */
+ export type DistanceFunc = (p: Point, a: Point, b: Point) => number;
+ 
+ /**
+  * Distance and index as returned by `MaxDistance()`.
+  */
+ type DistanceIndex = { distance: number, index: number };
+ 
+ /**
+  * Calculates the perpendicular distance between point `p` and the line intersecting points `a` and `b`.
+  * @param p point
+  * @param a line point
+  * @param b line point
+  * @returns distance
+  */
+ export const PerpendicularDistance: DistanceFunc = (p: Point, a: Point, b: Point) => {
+     let distance;
+ 
+     // horizontal line
+     if (a.x === b.x) distance = Math.abs(p.x - a.x);
+ 
+     // vertical line
+     else if (a.y === b.y) distance = Math.abs(p.y - a.y);
+ 
+     // sloped line
+     else {
+         const slope = (b.y - a.y) / (b.x - a.x);
+         const intercept = a.y - (slope * a.x);
+         distance = Math.abs(slope * p.x - p.y + intercept) / Math.sqrt(Math.pow(slope, 2) + 1);
+     }
+ 
+     return distance;
+ }
+ 
+ /**
+  * Calculates the shortest distance between point `p` and the line segment between points `a` and `b`.
+  * @param p point
+  * @param a line point
+  * @param b line point
+  * @returns distance
+  */
+ export const ShortestDistance: DistanceFunc = (p: Point, a: Point, b: Point) => {
+ 
+     // function to calculate the square of the distance between two points
+     const pointDistSq = (i: Point, j: Point) => Math.pow(i.x - j.x, 2) + Math.pow(i.y - j.y, 2);
+ 
+     let distance;
+     const lineLenSq = pointDistSq(a, b);
+ 
+     // line is actually just a point
+     if (lineLenSq === 0) distance = pointDistSq(p, a);
+ 
+     // line is a line
+     else {
+ 
+         // which endpoint is the point closer to?
+         const t = ((p.x - a.x) * (b.x - a.x) + (p.y - a.y) * (b.y - a.y)) / lineLenSq;
+ 
+         // closer to point A
+         if (t < 0) distance = pointDistSq(p, a);
+         
+         // closer to point B
+         else if (t > 1) distance = pointDistSq(p, b);
+ 
+         // somewhere in the middle
+         else {
+             distance = pointDistSq(p, {
+                 x: a.x + t * (b.x - a.x),
+                 y: a.y + t * (b.y - a.y)
+             });
+         }
+     }
+ 
+     // make sure we return the square root of the distance
+     return Math.sqrt(distance);
+ }
+ 
+ /**
+  * The default distance function if none is specified.
+  */
+ export let DefaultDistanceFunc = ShortestDistance;
+ 
+ /**
+  * Finds the data point in the curve that is furthest away from the straight line between the first and last points of the curve.
```

Comment:
Codacy has a fix for the issue: [Delete `·`](https://app.codacy.com/gh/icooper/curvereduce-js/pullRequest?prid=9960097)
```suggestion
 * Finds the data point in the curve that is furthest away from the straight line between the first and last points of the curve.
```